src/transformStateWithClones.js

```diff
@@ -7,7 +7,50 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let prevState = { ...state };
+
+  for (const action of actions) {
+    let stepResult = { ...prevState };
+
+    switch (action.type) {
+      case 'addProperties':
+        stepResult = {
+          ...stepResult,
+          ...action.extraData,
+        };
+
+        result.push(stepResult);
+
+        prevState = { ...stepResult };
+
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (stepResult[key]) {
+            delete stepResult[key];
+          }
+        }
+
+        result.push(stepResult);
```

Comment:
You push `stepResult` to an array in all cases.
Checklist:
- [DONT REPEAT YOURSELF] - If you perform same action in all switch cases - do it just once afterwards.

Reply:
But in `'clear'` case I need to push empty object, so I can't put it outside switch